### PR TITLE
fixes for TB60 (version .12 as in bug 1391020)

### PR DESCRIPTION
**Pull request by jobisoft**
Hi, these are the changes I made to quicktest, published as version .12 in bug [1391020](https://bugzilla.mozilla.org/show_bug.cgi?id=1391020). It is mostly working with TB60 again, [[CURSOR]] is still not working.

It also includes Jorg initial changes (Components.interfaces.nsILocalFile -> Components.interfaces.nsIFile)

### CODE REVIEW

**Comment by level420** [comment]
Great! Thank you @jobisoft works!

**Comment by jobisoft** [comment]
After this PR has been accepted, I will send another one for the pro version, which is not done yet (script support needs some testing)

File: chrome/content/quicktext.js [hunk 1]
```diff
@@ -109,21 +109,26 @@ var quicktext = {
...
-    if (document.getElementById("date-long"))
-      document.getElementById("date-long").setAttribute("label", this.mStringBundle.getFormattedString("date", [dateTimeService.FormatDate("", dateTimeService.dateFormatLong, timeStamp.getFullYear(), timeStamp.getMonth()+1, timeStamp.getDate())]));
-    if (document.getElementById("time-noseconds"))
-      document.getElementById("time-noseconds").setAttribute("label", this.mStringBundle.getFormattedString("time", [dateTimeService.FormatTime("", dateTimeService.timeFormatNoSeconds, timeStamp.getHours(), timeStamp.getMinutes(), timeStamp.getSeconds())]));
-    if (document.getElementById("time-seconds"))
-      document.getElementById("time-seconds").setAttribute("label", this.mStringBundle.getFormattedString("time", [dateTimeService.FormatTime("", dateTimeService.timeFormatSeconds, timeStamp.getHours(), timeStamp.getMinutes(), timeStamp.getSeconds())]));
+    var options = {};
+    options["date-short"] = { year: 'numeric', month: '2-digit', day: '2-digit' };
```
Comment (mkmelin) [inline chrome/content/quicktext.js]: nit: please prefer double quotes

File: chrome/content/quicktext.js [hunk 2]
```diff
@@ -109,21 +109,26 @@ var quicktext = {
...
-    if (document.getElementById("time-seconds"))
-      document.getElementById("time-seconds").setAttribute("label", this.mStringBundle.getFormattedString("time", [dateTimeService.FormatTime("", dateTimeService.timeFormatSeconds, timeStamp.getHours(), timeStamp.getMinutes(), timeStamp.getSeconds())]));
+    var options = {};
+    options["date-short"] = { year: 'numeric', month: '2-digit', day: '2-digit' }; 
+    options["date-long"] = { weekday: 'long', year: 'numeric', month: 'long', day: '2-digit' };
+    options["time-noseconds"] = { hour: '2-digit', minute: '2-digit' }; 
+    options["time-seconds"] = { hour: '2-digit', minute: '2-digit', second: '2-digit' };
+      
+    let fields = Object.keys(options);
+    for (let i=0; i<fields.length; i++) {
```
Comment (mkmelin) [inline chrome/content/quicktext.js]: here and elsewhere, space around operators, like for (let i=0; i < fields.length; i++)

File: components/wzQuicktext.js [hunk 3]
```diff
@@ -160,7 +160,7 @@ wzQuicktext.prototype = {
     this.mQuicktextDir = profileDir;
     this.mQuicktextDir.append("quicktext");
     if (!this.mQuicktextDir.exists())
-      this.mQuicktextDir.create(Components.interfaces.nsIFile.DIRECTORY_TYPE, 0755);
+      this.mQuicktextDir.create(Components.interfaces.nsIFile.DIRECTORY_TYPE, parseInt("0755", 8));
```
Comment (mkmelin) [inline components/wzQuicktext.js]: could use octal notation 0o755 instead

File: components/wzQuicktext.js [hunk 4]
```diff
@@ -602,20 +602,29 @@ wzQuicktext.prototype = {
...
-      var converter = Components.classes["@mozilla.org/intl/scriptableunicodeconverter"].createInstance(Components.interfaces.nsIScriptableUnicodeConverter);
-      converter.charset = "UTF-16";
-
-      var chunk = converter.ConvertFromUnicode(aData);
-      foStream.write(chunk, chunk.length);
+        //no longer works in TB60 - see bug 1391020 (https://bugzilla.mozilla.org/show_bug.cgi?id=1391020#c36)
+        //var converter = Components.classes["@mozilla.org/intl/scriptableunicodeconverter"].createInstance(Components.interfaces.nsIScriptableUnicodeConverter);
+        //converter.charset = "UTF-16"; 
+        //var chunk = converter.convertToByteArray(aData, {});
+
```
Comment (mkmelin) [inline components/wzQuicktext.js]: just remove the disfunct code

File: components/wzQuicktext.js [hunk 5]
```diff
@@ -602,20 +602,29 @@ wzQuicktext.prototype = {
...
+        //Polyfill for convertToByteArray, which no longer works with UTF-16
```
Comment (mkmelin) [inline components/wzQuicktext.js]: nit: space after // - so "// Polyfill for..."

File: components/wzQuicktext.js [hunk 6]
```diff
@@ -602,20 +602,29 @@ wzQuicktext.prototype = {
   {
     var foStream = Components.classes["@mozilla.org/network/file-output-stream;1"].createInstance(Components.interfaces.nsIFileOutputStream);
 
-    foStream.init(aFile, 0x02 | 0x08 | 0x20, 0664, 0);
+    foStream.init(aFile, 0x02 | 0x08 | 0x20,  parseInt("0664", 8), 0);
 
     // Unicode
     if (true)
     {
```
Comment (mkmelin) [inline components/wzQuicktext.js]: so more code to remove here? (Yes this was here before)

File: components/wzQuicktext.js [hunk 7]
```diff
@@ -602,20 +602,29 @@ wzQuicktext.prototype = {
...
+        //var chunk = converter.convertToByteArray(aData, {});
+
+        //Polyfill for convertToByteArray, which no longer works with UTF-16
+        let chunk = [];
+        for (let l=0; l<aData.length; l++) {
+            let c = aData.charCodeAt(l);
+            //fixed endianness
+            chunk.push( c & 0xFF );
+            chunk.push( (c >> 8) & 0xFF );
+        }
```
Comment (mkmelin) [inline components/wzQuicktext.js]: strange extra spaces here

**Comment by jobisoft** [comment]
made changes as requested

**Comment by mkmelin** [comment]
Merged, thx!

**Pull request merged**